extensions/elasticsearch/tests/MockElastic.h

```diff
@@ -0,0 +1,192 @@
+  }
+
+  rapidjson::Value addUpdateSuccess(rapidjson::Document::AllocatorType& alloc) {
+    rapidjson::Value item{rapidjson::kObjectType};
+    rapidjson::Value operation{rapidjson::kObjectType};
+    operation.AddMember("_index", "test", alloc);
+    operation.AddMember("_id", "1", alloc);
+    operation.AddMember("result", "updated", alloc);
+    item.AddMember("update", operation, alloc);
+    return item;
+  }
+
+  rapidjson::Value addUpdateError(rapidjson::Document::AllocatorType& alloc) {
+    rapidjson::Value item{rapidjson::kObjectType};
+    rapidjson::Value operation{rapidjson::kObjectType};
+    operation.AddMember("_index", "test", alloc);
+    operation.AddMember("_id", "1", alloc);
+    rapidjson::Value error{rapidjson::kObjectType};
+    error.AddMember("type", "document_missing_exception", alloc);
+    error.AddMember("reason", "[6]: document missing", alloc);
+    error.AddMember("index_uuid", "aAsFqTI0Tc2W0LCWgPNrOA", alloc);
+    error.AddMember("shard", "0", alloc);
+    error.AddMember("index", "index", alloc);
+    operation.AddMember("error", error, alloc);
+    item.AddMember("update", operation, alloc);
+    return item;
+  }
+
+  bool handlePost(CivetServer*, struct mg_connection* conn) override {
+    char request[2048];
+    size_t chars_read = mg_read(conn, request, 2048);
+
+    std::vector<std::string> lines = utils::StringUtils::splitRemovingEmpty({request, chars_read}, "\n");
+    rapidjson::Document response{rapidjson::kObjectType};
+    response.AddMember("took", 30, response.GetAllocator());
+    response.AddMember("errors", ret_error_, response.GetAllocator());
+    response.AddMember("items", rapidjson::kArrayType, response.GetAllocator());
+    auto& items = response["items"];
+    for (const auto& line : lines) {
+      rapidjson::Document line_json;
+      line_json.Parse<rapidjson::kParseStopWhenDoneFlag>(line.data());
+      if (!line_json.HasMember("index") && !line_json.HasMember("create") && !line_json.HasMember("update") && !line_json.HasMember("delete"))
+        continue;
+
+
+      rapidjson::Value item{rapidjson::kObjectType};
+      rapidjson::Value operation{rapidjson::kObjectType};
+
+      if (ret_error_) {
+        items.PushBack(addUpdateError(response.GetAllocator()), response.GetAllocator());
+      } else {
+        if (line_json.HasMember("update"))
+          items.PushBack(addUpdateSuccess(response.GetAllocator()), response.GetAllocator());
+        else
+          items.PushBack(addIndexSuccess(response.GetAllocator()), response.GetAllocator());
+      }
+    }
+
+    rapidjson::StringBuffer buffer;
+    rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+    response.Accept(writer);
+
+    mg_printf(conn, "HTTP/1.1 200 OK\r\n");
+    mg_printf(conn, "Content-length: %lu", buffer.GetSize());
+    mg_printf(conn, "\r\n\r\n");
+    mg_printf(conn, "%s", buffer.GetString());
+    return true;
+  }
+
+  bool ret_error_ = false;
+};
+
+class MockElastic {
+  struct CivetLibrary{
```

Comment:
this civet initializer is already defined in `TestServer.h` and `MockSplunkHEC.h`, it might be time to move it to a common place

Reply:
makes sense, changed it in https://github.com/apache/nifi-minifi-cpp/pull/1349/commits/f260dc059908a52310b14d6dbd09c9dbbe81fab5#diff-ae2a7d7ad1f15725f1e8101ea1a3c3d437942603e31e9ab66802a76b1641cad4R1